clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/common/AzureUtilities.groovy

```diff
@@ -112,6 +113,30 @@ class AzureUtilities {
     resourceId.split(PATH_SEPARATOR).last()
   }
 
+  // id = "/subscriptions/***-***-***/resourceGroups/***/providers/Microsoft.Network/networkInterfaces/nic1/ipConfigurations/ipconfig1"
+  static String getResourceNameFromId(String id) {
```

Comment:
I see now. I can change the name of the method to capture that we are looking at a "network" resource and not something else. Or we can use a fixed count of expected "/" to figure it out.